server/src/main/java/io/crate/planner/optimizer/rule/MoveConstantJoinConditionsBeneathNestedLoop.java

```diff
@@ -97,8 +96,8 @@ public LogicalPlan apply(NestedLoopJoin nl,
             // Push constant join condition down to source
             var lhs = resolvePlan.apply(nl.lhs());
             var rhs = resolvePlan.apply(nl.rhs());
-            var queryForLhs = constantConditions.remove(lhs.getRelationNames());
-            var queryForRhs = constantConditions.remove(rhs.getRelationNames());
+            var queryForLhs = constantConditions.remove(new HashSet<>(lhs.getRelationNames()));
+            var queryForRhs = constantConditions.remove(new HashSet<>(rhs.getRelationNames()));
```

Comment:
` constantConditions.remove` removes from `QuerySplitter.Context.allNames` which is created via RelationNameCollector.collect ie `LinkedHashSet`. We must create `LinkedHashSet` here as well, otherwise order of one set is static and order of new HashSet depends on incoming values.

Reply:
No, the Sets exclude the order from their equality:

```
jshell> var x = Set.of(1,2);
jshell> var y = new LinkedHashSet<Integer>();
y ==> []
jshell> y.add(2);
$3 ==> true
jshell> y.add(1);
$4 ==> true
jshell> x.hashCode() == y.hashCode();
$6 ==> true
jshell> x.equals(y);
$7 ==> true
```

Reply:
oh, yep, order is only for iterating. 👍